tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -462,15 +630,18 @@ def generate_gitlab_full_configuration(
     - compare_to: Override compare_to on change rules
     - return_dump: Whether to return the string dump or the dict object representing the configuration
     - apply_postprocessing: Whether or not to solve `extends` and `!reference` tags
+    - input_config: If not None, will use this config instead of parsing existing yaml file at `input_file`
     """
     if apply_postprocessing:
-        full_configuration = get_full_gitlab_ci_configuration(ctx, input_file)
+        full_configuration = get_full_gitlab_ci_configuration(ctx, input_file, input_config=input_config)
+    elif input_config:
+        full_configuration = deepcopy(input_config)
     else:
         full_configuration = read_includes(None, input_file, return_config=True)
 
     # Override some variables with a dedicated context
     if context:
-        full_configuration["variables"].update(context)
+        full_configuration['variables'] = full_configuration.get('variables', {}).update(context)
```

Comment:
Why is this change needed?

Reply:
I added this because sometimes there is no `variables` within the configuration (as in the current `deploy_containers_a7.yml`), however this is called only on the main configuration: I'll undo my changes